src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntTests.cs

```diff
@@ -360,6 +360,418 @@ static void verifyTypes(NamedTypeSymbol underlyingType, NamedTypeSymbol nativeIn
+            comp.VerifyDiagnostics(diagnostics);
+
+            comp = CreateCompilation(sourceB, references: new[] { ref2 }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(diagnostics);
+        }
+
+        [Fact]
+        public void Overloads_02()
+        {
+            var sourceA =
+@"public class A
+{
+    public void F1(System.IntPtr x) { }
+    public void F2(System.UIntPtr y) { }
+}";
+            var sourceB =
+@"class B1 : A
+{
+    public void F1(nint x) { base.F1(x); }
+    public void F2(nuint y) { base.F2(y); }
+}
+class B2 : A
+{
+    public void F1(nuint x) { }
+    public void F2(nint y) { }
+}
+class B3 : A
+{
+    public void F1(nint x) { base.F1(x); }
+    public void F2(nuint y) { base.F2(y); }
+}
+class B4 : A
+{
+    public void F1(nuint x) { }
+    public void F2(nint y) { }
+}";
+
+            var comp = CreateCompilation(sourceA, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+            var ref1 = comp.ToMetadataReference();
+            var ref2 = comp.EmitToImageReference();
+
+            var diagnostics = new[]
+            {
+                // (3,17): warning CS0108: 'B1.F1(nint)' hides inherited member 'A.F1(IntPtr)'. Use the new keyword if hiding was intended.
+                //     public void F1(nint x) { base.F1(x); }
+                Diagnostic(ErrorCode.WRN_NewRequired, "F1").WithArguments("B1.F1(nint)", "A.F1(System.IntPtr)").WithLocation(3, 17),
+                // (4,17): warning CS0108: 'B1.F2(nuint)' hides inherited member 'A.F2(UIntPtr)'. Use the new keyword if hiding was intended.
+                //     public void F2(nuint y) { base.F2(y); }
+                Diagnostic(ErrorCode.WRN_NewRequired, "F2").WithArguments("B1.F2(nuint)", "A.F2(System.UIntPtr)").WithLocation(4, 17),
+                // (13,17): warning CS0108: 'B3.F1(nint)' hides inherited member 'A.F1(IntPtr)'. Use the new keyword if hiding was intended.
+                //     public void F1(nint x) { base.F1(x); }
+                Diagnostic(ErrorCode.WRN_NewRequired, "F1").WithArguments("B3.F1(nint)", "A.F1(System.IntPtr)").WithLocation(13, 17),
+                // (14,17): warning CS0108: 'B3.F2(nuint)' hides inherited member 'A.F2(UIntPtr)'. Use the new keyword if hiding was intended.
+                //     public void F2(nuint y) { base.F2(y); }
+                Diagnostic(ErrorCode.WRN_NewRequired, "F2").WithArguments("B3.F2(nuint)", "A.F2(System.UIntPtr)").WithLocation(14, 17)
+            };
+
+            comp = CreateCompilation(sourceB, references: new[] { ref1 }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(diagnostics);
+
+            comp = CreateCompilation(sourceB, references: new[] { ref2 }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(diagnostics);
+        }
+
+        [Fact]
+        public void Overloads_03()
```

Comment:
>Overloads_03 [](start = 20, length = 12)

Could you please add a test (if we don't have one already) that shows that the following overloads are not permitted together in a type (either class or interface)?
```csharp
    public void F1(nint x) { }
    public void F1(System.IntPtr y) { }
```
and similarly for the unsigned versions.